src/mpi_c_bindings.f90

```diff
@@ -21,20 +21,22 @@ subroutine c_mpi_bcast_real(buffer, count, datatype, root, comm, ierror) bind(C,
             integer(c_int), optional, intent(out) :: ierror
         end subroutine
 
-        subroutine c_mpi_allgather_int(sendbuf, sendcount, sendtype, recvbuf, recvcount, recvtype, comm, ierror) bind(C, name="MPI_Allgather")
+        subroutine c_mpi_allgather_int(sendbuf, sendcount, sendtype, recvbuf, &
+                                       recvcount, recvtype, comm, ierror) bind(C, name="MPI_Allgather")
             import :: c_int, c_double
             integer(c_int), dimension(:), intent(in) :: sendbuf
-            integer(c_int), dimension(:, :) :: recvbuf
+            integer(c_int), dimension(*) :: recvbuf
```

Comment:
This is good, I think considering that we are going from `DescriptorArray` (which isn't used in `bind(C)`) to `UnboundedPointerToDataArray`